feat: set error.cause for tarn acquire connection error

s-sathish opened the pull request:
Log actual error from tarn during acquire connection before conversion. The underlying error can be anything.

It is better to surface out the actual error returned by tarn before converting it.

Also, not sure if instead of converting, why not send back the actual error itself ?

s-sathish committed:
Log actual error from tarn during acquire connection before convertion

s-sathish committed:
Add the actual error from tarn in the thrown exception incase of TimeoutError

s-sathish commented:
I have added the actual error from the tarn in the thrown exception incase of TimeoutError

s-sathish commented:
@kibertoad Ping

s-sathish commented:
@kibertoad Ping

owlas commented:
This would help to debug (if not solve) a lot of open issues, it's clearly a stumbling block for most developers running knex in prod:

https://github.com/knex/knex/issues/6034
https://github.com/knex/knex/issues/6051
https://github.com/knex/knex/issues/5696
https://github.com/knex/knex/issues/4932
https://github.com/knex/knex/issues/5658

s-sathish commented:
@tgriesser @kibertoad Ping

gothraven commented:
@owlas it's been years now .. why is this not moving forward ? I've spent three days already trying to understand whats going on when i'm following the docs to use transactions, but it keeps giving me `KnexTimeoutError` for no obvious reason.

myndzi committed:
Add test; tweak implementation

- Log acquire connection failures as warnings
- Log all acquire errors, not just timeouts
- Be deliberate about placement and stringification of Tarn
  TimeoutError messages
- Add test to verify that acquire error messages propagate

myndzi committed:
Update assertion to match new error message

tgriesser committed:
expose TimeoutError as error.cause on KnexTimeoutError, log warning

tgriesser committed:
Merge branch 'master' into log-acquire-connection-error

tgriesser committed:
Revert message change in test

s-sathish committed:
Merge branch 'master' into log-acquire-connection-error

tgriesser committed:
Merge branch 'master' into log-acquire-connection-error
